app/services/controller.py

```diff
@@ -0,0 +1,69 @@
+from fastapi import FastAPI,APIRouter,HTTPException
+from app.database.schema import Channel
+from bson.objectid import ObjectId
+channelsRouter = APIRouter() 
+from app.config.config import channel_DB
+from bson import ObjectId
+
+def fetch_channel():
+    """
+    Args : none
+
+    Fetching all the channels from the database
+
+    Return type : List
+    """
+    channels_list = []
+    for channel in channel_DB().find():
+        each_channel=dict(list(Channel(**channel)))
+        if (each_channel['type']=="public"):
+            channels_list.append(Channel(**channel))
+    return {'channels': channels_list}
```

Comment:
use the filter parameter in `find()` instead of doing this. like this: `channel_DB().find({"type": "public"})` you will have issues when you return it, so look at this link to figure out how to solve it: https://github.com/tiangolo/fastapi/issues/452

Reply:
Okay.We will change it